src/main/kotlin/uk/gov/justice/digital/hmpps/externalmovementsapi/entity/TemporaryAbsenceSeries.kt

```diff
@@ -0,0 +1,189 @@
+  val id: UUID = newUuid(),
+) {
+  @Size(max = 10)
+  @NotNull
+  @Column(name = "person_identifier", nullable = false, length = 10)
+  var personIdentifier: String = personIdentifier
+    private set
+
+  @Size(max = 6)
+  @NotNull
+  @Column(name = "prison_code", nullable = false, length = 6)
+  var prisonCode: String = prisonCode
+    private set
+
+  @Audited(targetAuditMode = NOT_AUDITED)
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "absence_type_id")
+  var absenceType: AbsenceType? = absenceType
+    private set
+
+  @Audited(targetAuditMode = NOT_AUDITED)
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "absence_sub_type_id")
+  var absenceSubType: AbsenceSubType? = absenceSubType
+    private set
+
+  @Audited(targetAuditMode = NOT_AUDITED)
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "absence_reason_id")
+  var absenceReason: AbsenceReason? = absenceReason
+    private set
+
+  @NotNull
+  @Column(name = "repeat", nullable = false)
+  var repeat: Boolean = repeat
+    private set
+
+  @NotNull
+  @Column(name = "release_at", nullable = false)
+  var releaseAt: LocalDateTime = releaseAt
+    private set
+
+  @NotNull
+  @Column(name = "return_by", nullable = false)
+  var returnBy: LocalDateTime = returnBy
+    private set
+
+  @Audited(targetAuditMode = NOT_AUDITED)
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "location_type_id")
+  var locationType: LocationType = locationType
+    private set
+
+  @Size(max = 36)
+  @Column(name = "location_id", length = 36)
+  var locationId: String? = locationId
+    private set
+
+  @NotNull
+  @Column(name = "accompanied", nullable = false)
+  var accompanied: Boolean = accompanied
+    private set
+
+  @Audited(targetAuditMode = NOT_AUDITED)
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "accompanied_by_id")
+  var accompaniedBy: AccompaniedBy? = accompaniedBy
+    private set
+
+  @Audited(targetAuditMode = NOT_AUDITED)
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "transport_id")
+  var transport: Transport? = transport
+    private set
+
+  @Audited(targetAuditMode = NOT_AUDITED)
+  @ManyToOne(fetch = FetchType.LAZY, optional = false)
+  @JoinColumn(name = "status_id", nullable = false)
+  var status: TapStatus = status
+    private set
+
+  @Column(name = "notes")
+  var notes: String? = notes
+    private set
+
+  @NotNull
+  @Column(name = "submitted_at", nullable = false)
+  var submittedAt: LocalDateTime = submittedAt
+    private set
+
+  @Column(name = "legacy_id")
+  var legacyId: Long? = legacyId
+    private set
+
+  @Size(max = 6)
+  @Column(name = "to_agency_code", length = 6)
+  var toAgencyCode: String? = toAgencyCode
+    private set
+
+  fun update(
+    personIdentifier: String,
+    request: TapApplicationRequest,
+    rdProvider: (ReferenceDataDomain.Code, String) -> ReferenceData,
+  ) = apply {
+    this.personIdentifier = personIdentifier
+    prisonCode = request.prisonId ?: "UNK"
```

Comment:
The NOMIS column is nullable but never has a null value. I think it's better to !! this rather than have the unknown fallback. That way we'll spot it if sync ever does send it null